airbyte-cdk/python/airbyte_cdk/sources/connector_state_manager.py

```diff
@@ -0,0 +1,44 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, List, Mapping, MutableMapping, Union
+
+from airbyte_cdk.models import AirbyteStateBlob, AirbyteStateMessage, AirbyteStateType
+
+
+class ConnectorStateManager:
+    """
+    ConnectorStateManager consolidates the various forms of a stream's incoming state message (STREAM / GLOBAL / LEGACY) under a common
+    interface. It also provides methods to extract and update state
+    """
+
+    # In the immediate, we only persist legacy which will be used during abstract_source.read(). In the subsequent PRs we will
+    # initialize the ConnectorStateManager according to the new per-stream interface received from the platform
+    def __init__(self, state: Union[List[AirbyteStateMessage], MutableMapping[str, Any]] = None):
+        if isinstance(state, MutableMapping):
+            self.legacy = state
+        elif self.is_valid_legacy_state(state):
+            self.legacy = state[0].data
+        else:
+            self.legacy = {}
```

Comment:
```suggestion
       if not state:
         self.legacy = {}
        if self.is_migrated_legacy_state(state):
           # Convert migrated legacy state to unmigrated legacy state for temporary backward compatibility
            self.legacy = state[0].data
        elif isinstance(state, MutableMapping):
           # Airbyte versions before X.X.X are not using per stream state, and state is stored in a MutableMapping
            self.legacy = state
        else:
            raise Exception("Unsupported state format")
```

Reply:
Agree with restructuring the condition. but will tweak the comments a little bit